[New Term] Python Built-in Functions: `issubclass()` (#1489)
Codecademy/docs: content/python/concepts/built-in-functions/terms/issubclass/issubclass.md

> ## **issubclass()**
> used to check if a class is a subclass of another class or not.

Reviewer: As for this very line, I have the following suggestion:

```suggestion
The **`issubclass()`** function returns `True` if a given [class](https://www.codecademy.com/resources/docs/python/classes) is a subclass of one or more classes.
```

### Syntax
`issubclass()` take two parameters:

`issubclass(class, classinfo)`
- **class** - class to be checked
- **classinfo** - class, type, or tuple of classes and types 

**Return type:** `True` if object is subclass of a class, or any element of the tuple, otherwise `False`.

> 

Reviewer: Let's start a new line and then insert a \```pseudo code snippet that looks like the following:

```
issubclass(class, candidateClasses)
```

> 

Reviewer: ```suggestion

The `issubclass()` function take two parameters:

- `class`- a validly defined class
- `candidateClasses` - a single class, type, or [tuple](https://www.codecademy.com/resources/docs/python/tuples) of classes and types

If any relationship exists, `True` is returned, otherwise `False`.

```

Reviewer: @cacaprog Looks like this suggestion remains from the last bit of feedback. Can we implement this?

> ### Example
> 

Reviewer: ```suggestion
## Example

The following example uses the `issubclass()` function to confirm whether the `myInfo` is a subclass of the `myAge` class:
```

For example sections, it's always important to lead a code example with 1-2 sentences 😄

```
class myAge:
  age = 36

class myInfo(myAge):
  name = "Homer"
  age = myAge

> x = issubclass(myInfo, myAge)
> print(x)

Reviewer: ```suggestion
print(issubclass(myInfo, myAge))
```

```

> Output: 
> `True`
> 

Reviewer: ```suggestion
This will print the following output:

```

Reviewer: And then let's add a \```shell output snippet with the result, like so:

\`\`\`shell
True
\`\`\`

> ### codebyte

Reviewer: ```suggestion
## Codebyte Example
```

> [codebyte language=python]
> class Car:
>     def __init__(carColor):
>       print('My car is ', carColor)
> 
> class Color(Car):
>     def __init__(self):
>       Car.__init__('white')
> 
> print(issubclass(Color, Car))
> print(issubclass(Car, Color))
> print(issubclass(Color, (list, Car)))
> print(issubclass(Car, (list, Car)))
> [/codebyte]
> 
> 

Reviewer: ```suggestion
```codebyte/python
class Car:
    def __init__(carColor):
      print('My car is ', carColor)

class Color(Car):
    def __init__(self):
      Car.__init__('white')

print(issubclass(Color, Car))
print(issubclass(Car, Color))
print(issubclass(Color, (list, Car)))
print(issubclass(Car, (list, Car)))
```

Let's add a closing \`\`\` to this as well 😄 

Also, what is this `list`?

## **issubclass()**

> The **`issubclass()`** function returns `True` if a given [class](https://www.codecademy.com/resources/docs/python/classes) is a subclass of one or more classes.

Reviewer: ```suggestion

The **`issubclass()`** function returns `True` if a given [class](https://www.codecademy.com/resources/docs/python/classes) is a subclass of one or more classes.
```

## Syntax

> `issubclass()` take two parameters:
> 

Reviewer: ```suggestion
```

For the "Syntax" section, let's place any non-code text below the \```pseudo block 😄

> `issubclass(class, candidateClasses)`

Reviewer: This need to be embedded in a \```pseudo code block, like so:

\`\`\`pseudo
issubclass(class, candidateClasses)
\`\`\`

> - `class` - a validly defined class
> - `candidateClasses` - a single class, type, or class, type, or [tuple](https://www.codecademy.com/resources/docs/python/tuples) of classes and types
> 
> if any relationship exists, `True` is returned, otherwise `False`.
> 
> 
> ## Example
> 

Reviewer: ```suggestion
```

> The following example uses the `issubclass()` function to confirm whether the `myInfo` is a subclass of the `myAge` class:

Reviewer: ```suggestion
## Example

The following example uses the `issubclass()` function to confirm whether the `myInfo` is a subclass of the `myAge` class:
```

> ```

Reviewer: ```suggestion
```py
```

Since this is Python code, we should mark the top of the code snippet as such. 😄

class myAge:
  age = 36

class myInfo(myAge):
  name = "Homer"
  age = myAge

print(issubclass(myInfo, myAge))
```

> This will print the following output:

Reviewer: ```suggestion
This will print the following output:

```

Same thing goes for code snippets: one empty line above and below it.

```shell
True
```

## Codebyte Example

> ```

Reviewer: ```suggestion
```codebyte/python
```

[codebyte language=python]

> class Car:
>     def __init__(carColor):
>       print('My car is ', carColor)
> 
> class Color(Car):
>     def __init__(self):
>       Car.__init__('white')
> 
> print(issubclass(Color, Car))
> print(issubclass(Car, Color))

Reviewer: ```suggestion
class Car:
  def __init__(car_color):
    print("My car is ", car_color)

class Color(Car):
  def __init__(self):
    Car.__init__('white')

print(issubclass(Color, Car))
print(issubclass(Car, Color))
```

When applied, these suggestions will keep this code snippet's format/style in sync with how it looks on Codecademy's Python courses.

[/codebyte]
```

> 
> 
> 
> 

Reviewer: ```suggestion
```

Too many empty trailing lines

---
Title: 'issubclass()'

> Description: 'Returns True <span style="color:red">if</span> a given [class](https://www.codecademy.com/resources/docs/python/classes) is a subclass of one <span style="color:red">or</span> more classes.'

Reviewer 2: ```suggestion
Description: 'Returns True if a given class is a subclass of one or more classes.'
```

Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Classes'
  - 'Functions'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

> The **`issubclass()` function returns `True` based on whether a given [class](https://www.codecademy.com/resources/docs/python/classes) is a subclass of one or more classes.
> 
> The **`issubclass()` function returns `True` based on whether a given [class](https://www.codecademy.com/resources/docs/python/classes) is a subclass of one or more classes.

Reviewer 2: ```suggestion

The **`issubclass()` function returns `True` based on whether a given [class](https://www.codecademy.com/resources/docs/python/classes) is a subclass of one or more classes.
```

## Syntax

```pseudo
issubclass(class, candidateClasses)
```

`issubclass()` take two parameters:

- `class` - A validly defined class.
- `candidateClasses` - A single class or type, or a [tuple](https://www.codecademy.com/resources/docs/python/tuples) of classes and types.

> If any relationship exists, `True` is returned, `False` otherwise.

Reviewer 3: ```suggestion
If `class` is a subclass of `candidateClasses` (direct or indirect), `True` is returned, `False` otherwise.
```

> - `class` - a validly defined class
> - `candidateClasses` - a single class or type, or a [tuple](https://www.codecademy.com/resources/docs/python/tuples) of classes and types

Reviewer 2: ```suggestion

- `class` - A validly defined class.
- `candidateClasses` - A single class or type, or a [tuple](https://www.codecademy.com/resources/docs/python/tuples) of classes and types.
```

> if any relationship exists, `True` is returned, otherwise `False`.

Reviewer 2: ```suggestion
If any relationship exists, `True` is returned, `False` otherwise.
```

> ## **issubclass()**
> ---
> Title: 'issubclass()'
> 
> Description: Returns `True` <span style="color:red">if</span> a given [class](https://www.codecademy.com/resources/docs/python/classes) is a subclass of one <span style="color:red">or</span> more classes.
> 
> Subjects:
>   - 'Computer Science'
>   - 'Data Science'
> 
> Tags:
>   - 'Classes'
>   - 'Functions'
> 
> CatalogContent:
>   - 'learn-python-3'
>   - 'paths/computer-science'
> ---

Reviewer: ```suggestion
---
Title: 'issubclass()'
Description: 'Returns True if a given class is a subclass of one or more classes.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Classes'
  - 'Functions'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---
```

Thanks for bring in this meta-data. 😄 

There's a few more things to do here:

- We no longer need `**issubclass()**` on line # 1.
- The `Description` should be surrounded by single quotes (`' '`).
  - We can't have extra special characters or inline code here.
- No empty lines between the meta-data fields.

> 

Reviewer: ```suggestion

The **`issubclass()` function returns `True` based on whether a given [class](https://www.codecademy.com/resources/docs/python/classes) is a subclass of one or more classes.

```

## Syntax

```pseudo
issubclass(class, candidateClasses)
```

`issubclass()` take two parameters:
- `class` - a validly defined class

> - `candidateClasses` - a single class, type, or class, type, or [tuple](https://www.codecademy.com/resources/docs/python/tuples) of classes and types

Reviewer: ```suggestion
- `candidateClasses` - a single class or type, or a [tuple](https://www.codecademy.com/resources/docs/python/tuples) of classes and types
```

if any relationship exists, `True` is returned, otherwise `False`.

## Example

> The following example uses the `issubclass()` function to confirm whether the `myInfo` is a subclass of the `myAge` class:

Reviewer: ```suggestion
The following example uses the `issubclass()` function to confirm whether `myInfo` is a subclass of the `myAge` class:
```

> ```pseudo

Reviewer: ```suggestion
```py
```

class myAge:
  age = 36

class myInfo(myAge):
  name = "Homer"
  age = myAge

print(issubclass(myInfo, myAge))
```

This will print the following output:
```shell
True
```

## Codebyte Example

> In the following example, the `issubclass()` function is used to check the relationship between the `Car` <span style="color:red">and</span> `Color` classes. While `Color` is a subclass of `Car`, the reverse is <span style="color:red">not</span> true:

Reviewer: ```suggestion
In the following example, the `issubclass()` function checks the relationship between the `Car` and `Color` classes. While `Color` is a subclass of `Car`, the reverse is not true:
```

Author: @Dusch4593 Hi Brandon, 
One question: how can I change font color without using html style? It's possible?

Reviewer: @cacaprog Rendering color in GitHub-flavored markdown is [an ongoing issue](https://github.com/github/markup/issues/1440#issuecomment-1387485185), it seems. For the most part, let's not attempt font color for the Docs entries at this time.
